code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,255 @@
+				P.owner = owner
+				var/list/owners = P.get_owners()
+				P.find_target(owners,assasination)
+				protection += P.target
+				objectives += P
+
+	var/datum/objective/sacrifice_ecult/SE = new
+	SE.update_explanation_text()
+	objectives += SE
+
+/datum/antagonist/e_cult/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, mob_override ? null : "Knowledge described in the book allowed you to overcome your clownish nature, allowing you to use complex items effectively.")
+	current.faction |= "e_cult"
+
+/datum/antagonist/e_cult/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, removing = FALSE)
+	current.faction -= "e_cult"
+
+/datum/antagonist/e_cult/roundend_report()
+	var/list/parts = list()
+
+	var/cultiewin = TRUE
+
+	parts += printplayer(owner)
+
+	//Removed sanity if(changeling) because we -want- a runtime to inform us that the changelings list is incorrect and needs to be fixed.
+	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
+
+	if(objectives.len)
+		var/count = 1
+		for(var/datum/objective/objective in objectives)
+			if(objective.check_completion())
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='greentext'>Success!</b></span>"
+			else
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='redtext'>Fail.</span>"
+				cultiewin = FALSE
+			count++
+
+	if(cultiewin)
+		parts += "<span class='greentext'>The heretic was successful!</span>"
+	else
+		parts += "<span class='redtext'>The heretic has failed.</span>"
+
+	parts += "<b>Knowledge Researched:</b> "
+
+	var/list/knowledge_message = list()
+	for(var/X in get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		knowledge_message += "[EK.name]"
+	parts += knowledge_message.Join(", ")
+
+	return parts.Join("<br>")
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/e_cult/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge += initialized_knowledge
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/e_cult/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/e_cult/proc/has_knowledge(datum/eldritch_knowledge/wanted)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/searched = X
+		if(initial(wanted.name) == searched.name)
```

Comment:
If you turn `researched_knowledge` into an associative list you can just do `researched_knowledge[initial(wanted.name)]` instead.

Reply:
it is fine. i dislike working with assoc lists since list is every data structure in existance shoved into little closet, and i get easily confused when dealing with these.

Reply:
done